source/common/network/cidr_range.cc

```diff
@@ -79,13 +79,20 @@ IpVersion CidrRange::version() const {
 }
 
 bool CidrRange::isInRange(InstanceConstSharedPtr address) const {
-  if (address == nullptr || length_ < 0 || address->type() != Type::Ip ||
-      address->ip()->version() != version()) {
+  if (address == nullptr) {
+    return false;
+  }
+
+  return isInRange(*address);
+}
+
+bool CidrRange::isInRange(const Instance& address) const {
+  if (length_ < 0 || address.type() != Type::Ip || address.ip()->version() != version()) {
     return false;
   }
   // Make an CidrRange from the address, of the same length as this. If the two ranges have
   // are the same, then the address is in this range.
-  CidrRange other = create(address, length_);
+  CidrRange other = create(address.ip()->addressAsString(), length_);
```

Comment:
The string / socket to shared pointer conversion can happen further along in the code but the way I'm doing it here is simplest in terms of number of line changes.